mlir/include/mlir/Target/LLVMIR/LLVMTranslationInterface.h

```diff
@@ -58,6 +60,16 @@ class LLVMTranslationDialectInterface
                  LLVM::ModuleTranslation &moduleTranslation) const {
     return success();
   }
+
+  /// Hook for derived dialect interface to translate or act on a derived
+  /// dialect attribute that appears on a function parameter. This gets called
+  /// after the function operation has been translated.
+  virtual LogicalResult
+  convertParameterAttr(LLVM::LLVMFuncOp function, int argIdx,
```

Comment:
Most of LLVM uses `unsigned` for indexes.

Reply:
What are you referring to?
I strongly believe in using `int` in absence of bit/mask operation.

Reply:
```
$ git grep " int " | wc -l
1161
$ git grep " unsigned " | wc -l
3449
```

also most of the former is in tests or tools.

That being said, I'm not a proponent of using `unsigned` (I'd rather use `int32/64_t` throughout) , but I am a proponent of consistency.

Reply:
How do you suggest to fix this: I see it as a bug and I am consistently using `signed int`  everywhere!
Should I sent patched updating every file for consistency or should we make sure new code use safer arithmetic patterns?

Reply:
I'd like to have some clarity on what's the convention... Has anything came out after https://discourse.llvm.org/t/rfc-coding-standards-prefer-int-for-regular-arithmetic-use-unsigned-only-for-bitmask-and-when-you-intend-to-rely-on-wrapping-behavior/52191 ?

Reply:
It does not seems like we'll be able to make anything a LLVM-wide policy here.

Reply:
MLIR side we could. I'm rather pro not making it worse though (e.g., I'd consider consistency when it's towards direction we'd want to go rather than where we ended up in undesirable position and keeping going down that route)